test/promise-test.js

```diff
@@ -0,0 +1,225 @@
+"use strict";
+
+var sinon = require("../lib/sinon.js");
+var { assert, refute } = require("@sinonjs/referee");
+
+async function getPromiseStatus(promise) {
+    var status = "pending";
+    var value = null;
+    promise
+        .then(function (val) {
+            status = "fulfilled";
+            value = val;
+        })
+        .catch(function (reason) {
+            status = "rejected";
+            value = reason;
+        });
```

Comment:
Is there any reason not to use `await` wrapped in `try/catch` instead of `then/catch` on the promise?

```suggestion
    try {
        status = "fulfilled";
        value = await promise;
    } catch (reason) {
        status = "rejected";
        value = reason; 
    }
```

Reply:
Yes, there is a good reason: We don't want to `await` the promise, because we want to know if the promise is still `pending`. I committed your suggestion, but then there's a test failure and I reverted.